OmegaExamples/src/generative-models.jl

```diff
@@ -0,0 +1,412 @@
+
+# ╔═╡ f2fd33e0-3451-44cb-9e69-8b627abf7729
+begin 
+	randomPair() = (randsample(1 ~ Bernoulli(0.5)), randsample(1 ~ Bernoulli(0.5)))
+	barplot(Dict(freqtable([randomPair() for _ in 1:1000])))
+end
+
+# ╔═╡ a066a0a9-e28f-45ef-bfba-ef23eaf75f96
+md"We see by examining this histogram that `[true, false]` comes out about 25% of the time. We may define the probability of a return value to be the fraction of times (in the long run) that this value is returned from evaluating the program – then the probability of `[true, false]` from the above program is 0.25."
+
+# ╔═╡ e045e940-57ea-408c-a1ac-aa17516a7523
+md"## Distributions in Omega"
+
+# ╔═╡ 59862489-9fd9-4e10-93a2-802d757bd0a8
+# About Distributions, ω and randsample
+
+# ╔═╡ 9a1afe94-6683-4cf7-bfb9-26cb5321de19
+md"## Constructing Marginal Distributions"
+
+# ╔═╡ 2dc5b682-aba3-4edb-a2cc-e10c9708b986
+# Constructing random variables by tranforming other random variables
+
+# ╔═╡ 98da911f-1338-4651-9e20-aaeab6c81c50
+md"## The rules of probability"
+
+# ╔═╡ 27fe69b6-ba83-4b79-8f30-df5841650e23
+md"While we can build the marginal distribution for even very complicated programs in Omega, we can also derive these marginal distributions with the “rules of probability”. This is intractable for complex processes, but can help us build intuition for how distributions work."
+
+# ╔═╡ 10e3b2f2-68ec-4c19-b473-469a47815205
+md"#### Product Rule"
+
+# ╔═╡ 8283c98e-23eb-4a34-b4f2-14722a4afbe7
+md"In the above example we take three steps to compute the output value: we sample from the first Bernoulli dsitribution, then from the second, then we make a list from these values. To make this more clear let us re-write the program as:"
+
+# ╔═╡ 0add56e4-adba-448b-a6bc-8c17284bda01
+begin
+	A = randsample(1 ~ Bernoulli())
+	B = randsample(1 ~ Bernoulli())
+	C = [A, B]
+end
+
+# ╔═╡ 547eeca0-202d-4b43-8fc8-7d81c054bd02
+md"We can directly observe (as we did above) that the probability of true for `A` is 0.5, and the probability of false from `B` is 0.5. Can we use these two probabilities to arrive at the probability of 0.25 for the overall outcome `C = [true, false]`? Yes, using the product rule of probabilities: The probability of two random choices is the product of their individual probabilities. The probability of several random choices together is often called the joint probability and written as P(A,B). Since the first and second random choices must each have their specified values in order to get `[true, false]` in the example, the joint probability is their product: 0.25.
+We must be careful when applying this rule, since the probability of a choice can depend on the probabilities of previous choices. For instance, we can visualize the the exact probability of `[true, false]` resulting from this program by defining a new random variable:"
+
+# ╔═╡ 1c260eff-636c-429d-992f-b7854f4398e1
+begin
+	a(ω) = (1 ~ Bernoulli())(ω)
+	b(ω) = (1 ~ Bernoulli(a(ω) ? 0.3 : 0.7))(ω)
```

Comment:
Using the same `ids` 1 and 1 will mean that this probably does not do what you think it is doing